src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -0,0 +1,441 @@
+/**
+ * @file string_encoding_test.cpp
+ * @author Jeffin Sam
+ * @author Mikhail Lozhnikov
+ *
+ * Tests for the StringEncoding class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/boost_backport/boost_backport_string_view.hpp>
+#include <mlpack/core/data/tokenizers/split_by_any_of.hpp>
+#include <mlpack/core/data/tokenizers/char_extract.hpp>
+#include <mlpack/core/data/string_encoding.hpp>
+#include <mlpack/core/data/string_encoding_policies/dictionary_encoding_policy.hpp>
+#include <boost/test/unit_test.hpp>
+#include <memory>
+#include "test_tools.hpp"
+#include "serialization.hpp"
+
+using namespace mlpack;
+using namespace mlpack::data;
+using namespace std;
+
+BOOST_AUTO_TEST_SUITE(StringEncodingTest);
+
+//! Common input for some tests.
+static vector<string> stringEncodingInput = {
+    "mlpack is an intuitive, fast, and flexible C++ machine learning library "
+    "with bindings to other languages. ",
+    "It is meant to be a machine learning analog to LAPACK, and aims to "
+    "implement a wide array of machine learning methods and functions "
+    "as a \"swiss army knife\" for machine learning researchers.",
+    "In addition to its powerful C++ interface, mlpack also provides "
+    "command-line programs and Python bindings."
```

Comment:
Might be worth making things a bit harder by adding some Unicode characters, just to make sure there aren't any issues with those.  (I doubt it, I think all the code is fine as is, but never hurts to check. :))

Reply:
You're right, I'll work on it.